core/lib/spree/core/importer/order.rb

```diff
@@ -52,48 +50,43 @@ def self.create_shipments_from_params(shipments_hash, order)
-                unit.order = order
-
-                # Spree expects a Inventory Unit to always reference a line
-                # item and variant otherwise users might get exceptions when
-                # trying to view these units. Note the Importer might not be
-                # able to find the line item if line_item.variant_id |= iu.variant_id
-                unit.variant_id = iu[:variant_id]
-                unit.line_item_id = line_items.select do |l|
-                  l.variant_id.to_i == iu[:variant_id].to_i
-                end.first.try(:id)
-              end
+            shipment = order.shipments.build
+            shipment.tracking       = s[:tracking]
+            shipment.stock_location = Spree::StockLocation.find_by_admin_name(s[:stock_location]) || Spree::StockLocation.find_by_name!(s[:stock_location])
+
+            inventory_units = s[:inventory_units] || []
+            inventory_units.each do |iu|
+              ensure_variant_id_from_params(iu)
+
+              unit = shipment.inventory_units.build
+              unit.order = order
+
+              # Spree expects a Inventory Unit to always reference a line
+              # item and variant otherwise users might get exceptions when
+              # trying to view these units. Note the Importer might not be
+              # able to find the line item if line_item.variant_id |= iu.variant_id
+              unit.variant_id = iu[:variant_id]
+              unit.line_item_id = line_items.select do |l|
+                l.variant_id.to_i == iu[:variant_id].to_i
+              end.first.try(:id)
+            end
 
-              # Mark shipped if it should be.
-              if s[:shipped_at].present?
-                shipment.shipped_at = s[:shipped_at]
-                shipment.state      = 'shipped'
-                shipment.inventory_units.each do |unit|
-                  unit.state = 'shipped'
-                end
+            # Mark shipped if it should be.
+            if s[:shipped_at].present?
+              shipment.shipped_at = s[:shipped_at]
+              shipment.state      = 'shipped'
+              shipment.inventory_units.each do |unit|
+                unit.state = 'shipped'
               end
+            end
 
-              shipment.save!
-
-              shipping_method = Spree::ShippingMethod.find_by_name(s[:shipping_method]) || Spree::ShippingMethod.find_by_admin_name!(s[:shipping_method])
-              rate = shipment.shipping_rates.create!(:shipping_method => shipping_method,
-                                                     :cost => s[:cost])
-              shipment.selected_shipping_rate_id = rate.id
-              shipment.update_amounts
+            shipment.save!
 
-            rescue Exception => e
-              raise "Order import shipments: #{e.message} #{s}"
-            end
+            shipping_method = Spree::ShippingMethod.find_by_name(s[:shipping_method]) || Spree::ShippingMethod.find_by_admin_name!(s[:shipping_method])
```

Comment:
Same goes for here. Why search for the shipping method by searching for the shipping method by searching for the shipping method name?